### PR TITLE
naive alsc

### PR DESCRIPTION
**Pull request by eguchi1904**
## Check list

- [ ] I wrote [CHANGELOG.md](./CHANGELOG.md) if the pull request adds/modifies features.

## Summary

simplified version of [alsc algorithm in libcamera](https://github.com/raspberrypi/libcamera/blob/3fad116f89e0d3497567043cbf6d8c49f1c102db/src/ipa/raspberrypi/controller/rpi/alsc.cpp#L772) with the following omitted
 - dynamic lens shading table calculation based on estimated colour templature
 - "Adaptive ALSC Algorithm"

### CODE REVIEW

**Comment by eguchi1904** [comment]
@ishiy1993 @KeenS  master branch merge is complete and this PR is now ready for review.

File: actfw_core/unicam_isp_capture.py [hunk 1]
```diff
@@ -31,8 +38,34 @@
...
+    scale: Tuple[float, float]
+    crop: Tuple[int, int]
+
+
+V1_SENSOR_SIZE = (2592, 1944)
+V1_UNICAM_MODES: List[CameraMode] = [
+    CameraMode(size=(2592, 1944), scale=(1.0, 1.0), crop=(0, 0)),
+    CameraMode(size=(1920, 1080), scale=(1.0, 1.0), crop=(336, 432)),
+    CameraMode(size=(1296, 972), scale=(2.0, 2.0), crop=(0, 0)),
+    CameraMode(size=(640, 480), scale=(2.0, 2.0), crop=(656, 492)),
```
Comment (ishiy1993) [inline actfw_core/unicam_isp_capture.py:59]: Where do crop values come from?

File: actfw_core/unicam_isp_capture.py [hunk 2]
```diff
@@ -31,8 +38,34 @@
...
 
-V1_UNICAM_SIZES: List[Tuple[int, int]] = [(2592, 1944), (1920, 1080), (1296, 972), (640, 480)]
-V2_UNICAM_SIZES: List[Tuple[int, int]] = [(3280, 2464), (1920, 1080), (1640, 1232), (640, 480)]
+MAX_LS_GRID_SIZE = 0x8000
+LS_TABLE_W = 16
+LS_TABLE_H = 12
+
+
+# correspond to [libcamera's CameraMode](https://github.com/raspberrypi/libcamera/blob/3fad116f89e0d3497567043cbf6d8c49f1c102db/src/ipa/raspberrypi/controller/camera_mode.h#L19) # noqa: E501, B950
+@dataclass(frozen=True, eq=True)
```
Comment (ishiy1993) [inline actfw_core/unicam_isp_capture.py:47]: Python 3.6 does not support dataclass. Something wrong may happen in jetson.

https://docs.python.org/ja/3.10/library/dataclasses.html

Comment (eguchi1904) [reply actfw_core/unicam_isp_capture.py:47]: actfw-core depends `dataclasses` package in python 3.6 environment, so it seems ok.
https://github.com/Idein/actfw-core/blob/master/pyproject.toml#L38

also dataclass is already used [here](https://github.com/Idein/actfw-core/blob/3587d964011007a1e153ac77941a81161f8403ea/actfw_core/_private/schema/agent_app_protocol.py#L26)

File: actfw_core/unicam_isp_capture.py (hunk 1 above)
Comment (eguchi1904) [reply actfw_core/unicam_isp_capture.py:59]: There must be a more appropriate source, but I refer to picamera document.
https://picamera.readthedocs.io/en/release-1.13/fov.html#sensor-modes
> Modes with partial FoV capture from the center of the sensor.

**Review by ishiy1993 — APPROVED** [review]

File: actfw_core/unicam_isp_capture.py (hunk 1 above)
Comment (eguchi1904) [reply actfw_core/unicam_isp_capture.py]: >  I refer to picamera document.
> https://picamera.readthedocs.io/en/release-1.13/fov.html#sensor-modes

I looked it again and noticed that the parameters for v1 640x480 don't match the table.
https://github.com/Idein/actfw-core/pull/67/commits/09e0a0ae607d394f21bb7c7ec8aaebe9bf03f494

**Review by ishiy1993 — APPROVED** [review]

**Comment by eguchi1904** [comment]
:pray:

**Pull request merged**